---
layout: page
title: "Entry8  - 11/12-16"
description: "Entry8"
---
{% include JB/setup %}

**Entry 8**
After some (well, a lot...) hassle the Kallisto-DE pipeline is now complete. After running Kallisto, I used [tximport](https://bioconductor.org/packages/release/bioc/html/tximport.html) to convert my transcript-counts to gene-level counts for subsequent import into DE-analysis pipelines. The point were I got stuck at, was the annotation-part during the transcript-ID to gene-ID conversion. The transcript-IDs that I got out from Kallisto was in another format as the expected following the manual of tximport. This is of course a result of the transcriptome-reference I had chosen to start with and not of Kallisto software per se. The result was that the IDs didn't match. After looking at the different naming-conventions, the problem was the following:

I wrote a small R-script to get rid of the last part of each ID (after the "."). Caveat here, after searching around on the issue, I do not think that I loose information removing this part, and that the remaining gene-ID is sufficient. 

I begin by looking at the samples that at least in theory should display a clear divergence in transcriptional abundances of certain transcripts, i.e. non-infected vs virus-infected (highest amount).  After running Kallisto and aggregating the transcript-counts to gene-level counts, I ran the DESeq2 pipeline and produced the following MA-plot:

Kallisto-MA (Medium vs Virus)
![Psuedoalignment]({{ site.url }}/assets/images/Medium_vs_0.2MOI_Kallisto.png)

As expected we do see some clear DE effects here. Summary statistics:

    out of 33679 with nonzero total read count
    adjusted p-value < 0.05
    LFC > 0 (up)     : 836, 2.5% 
    LFC < 0 (down)   : 502, 1.5% 
    outliers [1]     : 0, 0% 
    low counts [2]   : 15671, 47% 
    (mean count < 8)

Lets move on to a mapped based approach with the same samples. For this I used STAR to map against the latest human reference genome from Ensembl, followed by HTSeq-count in union mode to assign reads to gene-counts. I imported the gene-counts and run the DESeq2 pipline in the same manner as for the Kallisto output. MA-plot: 

STAR_HTSeq-MA (Medium vs Virus)
![Psuedoalignment]({{ site.url }}/assets/images/Medium_vs_0.2MOI_mapped.png)

Just a quick look at the MA-plot I note similarity between the methods. However, looking at the summary statistics do show a significant difference:

Summary statistics:

    out of 19403 with nonzero total read count
    adjusted p-value < 0.05
    LFC > 0 (up)     : 902, 4.6% 
    LFC < 0 (down)   : 538, 2.8% 
    outliers [1]     : 0, 0% 
    low counts [2]   : 6771, 35% 
    (mean count < 10)

The real big difference here is the dimension of the DESeqDataSet. I have ~20k genes from the mapping pipeline, and ~34k genes from the kallisto pipeline. Why? After checking every step in my pipeline, I realize that my input from the kallisto pipeline is ~61k genes, while it is ~33k from the alignment-based pipeline. 

Why? I think that the real underlying cause here is the difference in annotation, i.e. the transcriptome I used to shred  into k-mers for the kallisto pipeline, vs the genome-reference that I used to mapp against with STAR. 
In order to shrink the memory size of the DESeqDataSet and remove genes with nearly zero counts, I perform a pre-filtering via:

    dds = dds[rowSums(counts(dds))>1,]

i.e. each gene that have 0 counts across all samples is removed from the dataset. Performing this operation leads to substantiation differences between the data-sets. For the mapping-set:

    Pre-filtering removed: 13880 genes

For the kallisto-set:

    Pre-filtering removed: 27323 genes

There are apparently many identified genes in the kallisto-set that have very low expression values. Still, I end up with a data set that, after filtering, consists of +~50% geneIDs. So, I went out on a quest to try to get insight into what these genes that are not present in both data sets consists off. Are they only very lowly expressed geneIDs that are a artifact of the pipelines and do not represent anything "important"?
Iterating through all geneIDs in the both data sets I filtered out 15310 IDs that were not included in both data sets. I created an index of these IDs and a new data set which contained all counts for these genes. 

    diff_index = setdiff(rownames(kallisto_dds), rownames(STAR_dds))
    unique_kallisto = counts(kallisto_dds)[rownames(counts(kallisto_dds)) %in% diff_index,]

Since the setdiff-function is nonsymmetric, you need to do the opposite direction aswell. Turns out that the vast majority of genes that are not included in both data sets derive from the kallisto data set, i.e. almost each gene ID that is included in the STAR-mapped data set, is also included in the kallisto data set. I added a column to my new data set with unique_kallisto geneIDs where I put the total amount of counts for each geneID:

     unique_kallisto = cbind(unique_kallisto, Total=rowSums(unique_kallisto))

And then sorted the set based on the total counts. Looking at the top50 geneIDs with most total counts, I have geneIDs with VERY many counts. Some genes that are highly expressed in all samples, and some genes that display a very clear differential expression where they are highly expressed in two samples (non-infected or infected) and very lowly expressed in the counterpart samples. I give you a quick look:

    GeneID          sample1 sample2 sample3 sample4  total
    ENSG00000236884       0       0   60876   46596   107472
    ENSG00000265972   16683   41361   14707   40383   113134
    ENSG00000198938   50076   18698   38588   21975   129337
    ENSG00000224103    4073   45370   49186   38884   137513
    ENSG00000277027   15812   40515   52420   31586   140333
    ENSG00000228080   66117   77973      28      22   144140
    ENSG00000260314   70212   28066   39195   16485   153958
    ENSG00000198804   58917   39593   54816   54307   207633
    ENSG00000206308   91851   77568   78225   65962   313606
    ENSG00000230726   91851   77568   78225   65962   313606
    ENSG00000234794   91851   77568   78225   65962   313606
    ENSG00000225840   23025   56838  137574  144151   361588
    ENSG00000280614   39597   90841  184992  242036   557466
    ENSG00000280800   39597   90841  184992  242036   557466
    ENSG00000281181   39597   90841  184992  242036   557466
    ENSG00000262406  215377  268520   60866   76147   620910
    ENSG00000274012  127645  417601  375106  412620  1332972
    ENSG00000211459  420997  413046  424784  570795  1829622
    ENSG00000276168  304515  979462  634011  656311  2574299
    ENSG00000210082 3498040 3125110 3179430 4275200 14077780

So, are these genes really unique for kallisto and not at all detected in the mapping approach? In order to figure this mystery out, I started with the tedious process of manually going through the genes with high counts-values. Following the same table as above:


    Known protein coding (MHC)
    Known protein coding
    mitochondrially encoded
    Known protein coding (MHC)
    ncRNA
    Known protein coding (MHC)
    mitochondrially encoded
    Known protein coding (MHC)
    Known protein coding (MHC)
    Known protein coding (MHC)
    Known processed pseudogene
    Known lincRNA
    Known lincRNA
    Novel lincRNA
    matrix metallopeptidase
    ncRNA
    mitochondrially encoded
    ncRNA
    mitochondrially encoded


This is a subset of the genes that I looked through. There is a apparent theme among the genes, either they are located on a MT chromosome (mitochondrally encoded), ncRNAs or, in the case of "known protein coding", they almost almost exclusively belong to the "major histocompatibility complex" family.  The fact that the top hits corresponds to mitochondrial och ncRNAs are of no surprise, and my theory is that these are not included in the references I used for STAR mapping (I know that the mitochondrally are not, which is deliberately since they are of no biological interest to us). However, of special interest is the geneIDs that show significant difference between samples. These genes all seems to belong to the MHC family. 

All of the known protein coding genes are flagged as "There is no ungapped mapping of this gene onto the GRCh37 assembly." That would prevent the annotation of a full length model. This is indeed the assembly that was used for STAR mapping. Further, after some searching I realize that all of the protein coding parts that are elusive to my Kalliso data set derives from haplotype variants, e.g. the "HSCHR6_MHC". From the Ensembl website regarding the primary assembly that was used during STAR mapping: 
"Primary assembly contains all toplevel sequence regions **excluding haplotypes
and patches**. This file is best used for performing sequence similarity searches
where patch and haplotype sequences would confuse analysis."

It's generally a good practice to exclude non-unique alignments when looking at differential expression. It the align-algorithm stumble upon the MHC genes that consists of several versions, it will not be able to tell if these are variants of the same region and if the read maps to this location, will be flagged as multiple mapping and excluded. It's generally recommended that these regions are only included if there are special interest for them in the biological question at hand. 

In summary, all the kallisto_unique IDs are absent from the primary assembly reference I used during STAR mapping. 

Ok, that was a lot of work to end up with the conclusion that everything seems to be in order. The really question that still remains to be answered is how to access the performance of the genes, and what is the "true" answer. My plan is the following, (1). Look more into detail of the transcripts that are present in both data sets and how they correlate to each other. (2). In the quest of trying to get somewhat of an "benchmark" parameter, I will go through the literature and create a list of genes that one **would assume to be up-regulated during viral infection**. "Marker"-genes one might say. Then specifically look up these genes in the different data sets. 
