---
layout: page
title: "Entry  - 30/11-16"
description: "Entry2"
---
{% include JB/setup %}

### Comments from group meeting and some points for consideration
We have decided upon the following points:

* I will send the group the article that I linked in entry 1, were the author makes the suggestion that one should use aggregation of transcripts to obtain gene-level counts even though gene-level counting is the goal from the beginning. I also suggested that everyone should read the [kallisto paper](http://www.nature.com/nbt/journal/v34/n5/full/nbt.3519.html). I think this should be a good starting point to have a proper idea of the tools that we are using. I have never used any alignment-independent tool before myself.
*The following pipeline map from our .ppt show the different routes one can take when doing differential expression:


![Pipelines]({{ site.url }}/assets/images/workflow.png)


* This also highlights the tools that we have decided to use for the analysis. Note that we plan to make use of two different tools for alignment-independent analysis (Kallisto, Sailfish). My take is that since the alignment-independent methods are less standard, especially for gene-level counting, it would be of interest to look at more than one such tool.

* The workflow above does not include the R-package ["Sleuth"](https://github.com/pachterlab/sleuth). This package make use of the Kallisto algoritm and performs differential expression from the output of "bootstraped" Kallisto (recommended approx 100 bootstraps from the [authors](https://liorpachter.wordpress.com/2015/08/17/a-sleuth-for-rna-seq/). Our plan was to include this package in our analysis. As i write this right now, I realize that the output from sleuth are transcript-level DE. But there might be some way to efficiently get the output to gene-level, I will have to look this up.

* We will focus our project solely on gene-level counting to begin with. One could also try to look at isoform abundances in both an alignment-dependent vs independent approach, but in this stage I think it's best to limit the project. Eventual this could be added at a later stage. I also think that it will be much harder to assess the final outcome from isoform differential expression data, as this type of data inherently are hard to compute and comes with a overall high uncertainty to begin with.

* I will make a project folder on the course uppmax folder /proj/g2016025/xxxx where I will upload the fastq-files. I will make a directory structure with separate folders for data-files, scripts, logs etc.


