---
layout: page
title: "Entry10  - 28/12-16"
description: "Entry10"
---
{% include JB/setup %}

**Added Salmon to the mix**

I continued by performing the same procedure outline earlier with the quantification tool [Salmon](https://combine-lab.github.io/salmon/). Which like Kallisto is an alignment-independent tool based on what the authors like to call "quasi-mapping".
After running Salmon, I did the same procedure as I did with Kallisto, i.e. using the bioconductor package tximport to aggregate the estimated transcript counts to gene-level estimations and then running DESeq2 for DE analysis.

Exploratory analysis was performed with a range of different plots and statistics (just as I did before), and I will just paste some of the plots below:

P-value distributions look healthy:
![pvalue-histo]({{ site.url }}/assets/images/p-value-histo-salmon.png)

So does the dispersion variance:

![disp-variance]({{ site.url }}/assets/images/disp_plots_salmon.png)

Labeled MA-plots (strikingly similar to Kallisto output):

![MA-plots]({{ site.url }}/assets/images/MA_plots_salmon_kallisto.png)
 
**Comparisons STAR/HTSeq, Kallisto, Salmon**

VennDiagram created to visualize similarity between genes that were flagged as differentially expressed (FDR level 5%):

![venn-polyl]({{ site.url }}/assets/images/Venn_med_vs_polyl.png)


![venn-virus]({{ site.url }}/assets/images/Venn_med_vs_Virus.png)

There is overall high concordance regarding the genes that are flagged as significant DE genes. There is specifically high concordance between the two alignment-independent pipelines, only a small subset of genes flagged from one of the pipeline are
"missing" in the other. More importantly, the most DE genes shows higher concodrance between all methods, and the gene sets diverge when I look at the genes with smaller logfoldchanges between samples.
This is also evident from the correlations plots below.

Correlations plots on count values used as input to DESeq2 between the 3 different methods (log transformed data after addition of a +1 psuedocount for visualization purposes):

Medium sample
![corr1]({{ site.url }}/assets/images/cor_plot_sample1.png)

       			  Salmon  Kallisto STAR/HTSeq
	Salmon     1.0000000 0.9937522  0.8961659
	Kallisto   0.9937522 1.0000000  0.9006621
	STAR/HTSeq 0.8961659 0.9006621  1.0000000


Polyl sample
![corr2]({{ site.url }}/assets/images/corr_plot_sample15.png)

             	  Salmon  Kallisto STAR/HTSeq
	Salmon     1.0000000 0.8983435  0.8408198
	Kallisto   0.8983435 1.0000000  0.9335632
	STAR/HTSeq 0.8408198 0.9335632  1.0000000

Virus sample
![corr3]({{ site.url }}/assets/images/corr_plot_sample6.png)

            	  Salmon  Kallisto STAR/HTSeq
	Salmon     1.0000000 0.9169367  0.8592406
	Kallisto   0.9169367 1.0000000  0.9350738
	STAR/HTSeq 0.8592406 0.9350738  1.0000000



Correlation in the final output of log2FoldChange:

For the comparison Medium vs Polyl:

![corr4]({{ site.url }}/assets/images/LFC-corr-med-vs-polyl.png)

              	  Salmon  Kallisto STAR/HTSeq
	Salmon     1.0000000 0.9886532  0.9093397
	Kallisto   0.9886532 1.0000000  0.9125289
	STAR/HTSeq 0.9093397 0.9125289  1.0000000

For the comparison Medium vs Virus:

![corr4]({{ site.url }}/assets/images/LFC-corr-med-vs-Virus.png)

              	  Salmon  Kallisto STAR/HTSeq
	Salmon     1.0000000 0.9885838  0.9200198
	Kallisto   0.9885838 1.0000000  0.9198928
	STAR/HTSeq 0.9200198 0.9198928  1.0000000

Notice the very high correlation between the two alignment-independent methods compared to STAR/HTseq. Not that surprising given that the same transcriptome was used as reference for these two methods.
Which gives them more of the same "playing field", compared to using STAR/HTseq with the human genome reference. However, it's interesting to note that the pipelines differ considerably more in the polyl and virus samples
than the medium samples. I don't know why, but the different algorithms of Kallisto/Salmon seems to decipher the polyl/virus-affected transcripts in different ways. Looking at estimated counts,
I took a look geneIDs that showed the largest difference between Kallisto/Salmon, but couldn't find any general pattern that would give me answer to the observed discrepancy. 
E.g. largest difference was observed for the gene [PLLP](http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000102934;r=16:57248547-57284687) , a gene that was identified in all three pipelines,
but with significantly larger estimated count values in the Salmon pipeline. Without looking deeper into the algorithmic behaviour, one could imagine a transcript (almost all estimated counts derives from one and same transcriptome the two pipelines in this case),
that display high similarity to another transcript of another gene, and that the assignment is reversed for this gene. However, when looking at the most DE genes, Kallisto/Salmon do display high similarity, and almost every geneID of 
the top100 most DE genes in the output from one of the pipelines are indeed included in the top100 output from the other pipeline. 