---
layout: page
title: "Entry  - 29/11-16"
description: "This is a test page"
---
{% include JB/setup %}

### Project background and setup
The idea behind this project is based on the notion I recently have stumbled upon regarding alignment-independent quantification of RNA-seq data. From various sources I've heard the argument that you might as well use alignment-independent based counting of transcripts instead of the alignment-dependent counterpart, going even further, I've heard the argument that these tools could even outperform  the alignment-based ones. This paper https://f1000research.com/articles/4-1521/v2 make the claim that aggregation of alignment-independent generated transcript to gene-level counts is the way to go.

The output from these alignment-free tools are counts per transcript. Hence, one can aggregate this output to gene-level counts, and use these count-values for DE analysis (e.g. DESeq2).

Below I will give a few initial comments on why it at least in theory might be worth looking at transcript-abundance even though the biological question at hand would be answered by gene-level quantification and when the isoform-profile per se is not of interest.

So sum it up, the project main goal to look at is the following:

* What are the difference between the output for gene-level quantification between an alignment-based tool (HTSeq-count) vs an alignment-free tool (Kallisto)

The dataset that we will use consists of 18 samples of monocyte-derived dendritic cells (moDCs) that have been either treated with virus (H1N1 influenza virus) or Polyl:C (dsRNA mimic) or non-treated. The question at hand is to look at differential gene expression (at the gene-level) to see how the viral infection affects the transcriptomic profile of the cells. We will also look at the eventual difference between "real" viral infection vs the "psuedo" vial infection that polyl:C could be seen to represent (this molecule is often used in biomedical research to stimulate an immune-response without actual killing the cells in the way that viruses do). The idea is to use both an alignment-based approach and an alignment-independent approach to get gene-level counts and then use DESeq2 for DE analysis. Some kind of comparison between the output should be made (VennDiagram etc?).


### Comments regarding gene/transcript-level counting and why you would even bother
The transcription of a single gene can result in multiple different transcripts via alternative spicing events. As these different transcripts that originate from the same gene often share some of the exons, they display less sequence uniqueness between each other than the sequence uniqueness that you typically see between different genes. Hence, the quantification procedure on transcript-level faces a harder problem - which transcript do the read originate from?

Accordingly one should generally look at the gene-level if your biological question is answered at this level of resolution.

Differential expression (DE) analysis at the gene-level is well established today with many tools available. In comparison, DE analysis at the isoform resolution have not reached the same level of usability and robustness.

**Gene-level quantification** can be achieved in principle in two alternative ways. (1) The alignment based approach would be to take your reads, align them to a reference genome and for each read that align to a gene then assign a gene-count.  (2) The alignment-independent approach would be to instead use an algorithm that looks for the most likely transcript for each read. When you have decided which transcript the read originate from, you would then collect all transcripts that belong to a given gene and aggregate the counts to get the total count for the specific gene.
Both approaches arrives at gene-level quantification.

**Transcript-level quantification** could also be either alignment-based or alignment-independent. The alignment-based approaches either uses a direct-exon counting approach or a somewhat more complicated algorithm that uses a maximum-likelihood estimation probability to assign a read to a particular transcript. In a effort for a very simple explanation, the main idea is that you have a set of isoforms for a gene, and if you know the distribution of these isoforms you could assign a probability for a match between the read and the different isoforms based on the isoform abundance. But as we realize, this abundance value is unknown to us. The abundances value could be calculated if we have knowledge about the probability of which isoform the read originates from, but alas, we don't now this probability either.  The Cufflinks algorithm tries to solve this dilemma by iteration. It starts with some kind of probability/abundance value to have a string point. Then, based on the compability between read-isoform, it make guesses about the probabilities that the reads comes from each isoform. Once we have this probability, it makes a recalculation of the abundances based on this new value. We get a new abundance value, and they it makes new probabilities based on this new abundance value. Iteration of the process proceeds until convergence.
The other alternative would be to once again use alignment-independent methods to count the transcripts, in the same manner as stated above in the "Gene-level" quantification. I will in a later post go through in more detail how Kallisto tries to solve this issue of alignment-independent quantification.

So, while it is evident that one would need transcript-level identification when looking at differential isoform-expression. Why would one even consider using transcript-counting tools and then aggregate to the gene-level when answering biological questions at the gene-level? Why not only use the simple counting approaches to begin with if you are not interested in isoform information? Since the transcript-quantification tools tries to solve an inherited difficult problem, it might not make sense to actual use these tools for gene-level quantification.

The simple counting approaches (OBS, here I talk about simple counting vs transcript-counting, and the latter could be achived via both alignment-dependent and alignment-independent methods), do come with some problematic characteristics. I will include a picture from the [Cufflinks paper](http://www.nature.com.focus.lib.kth.se/nbt/journal/v31/n1/full/nbt.2450.html) to illustrate this:

![Simple count-based quantification - picture taken from doi:10.1038/nbt.2450]({{ site.url }}/assets/images/Cufflinks_paper_simple_count_problem.PNG)

The simple approaches sums the read that match the exons of the transcripts. Typically this can be achieved via the "Exon-union" or "Exon-intersection" modes. However, both these modes may result in an incorrect estimation of differential expression which is highlighted in the picture above. The true expression change is estimated by taking into consideration the length of the isoform, why is this important? When we perform RNA-seq experiments, you typically have reads that are (much) shorter the transcripts they originate from. Since the transcripts are "chopped up" in this manner, there is a higher probability to get a read from a longer transcript. We see in the picture a few different potential outcomes that all lead to a wrongly estimated DE result.While this problematic would indeed go away if we were sequencing full-length transcripts, that is far from the case with the technologies used today, and, accordingly we have to deal with this issue.
