---
layout: page
title: "Entry11  - 03/01-17"
description: "Entry11"
---
{% include JB/setup %}

**Some conclusions and poster creation**

Beginning to wrap this project up. I think I have more or less stated my conclusions in the prior posts but it might be needed for some kind of summary.
We have been looking at monocyte-derived dendritic cells (moDCs), where we had 3 different conditions; Non-treated (Medium), Virus-infected and Polyl:C-treated.
From these we were interested in two comparisons, what transcriptional effect does Virus-infection have in moDCs? and, what transcriptional effect does polyl:C have in moDCs?
These questions were answered via differential expression analysis. I will give a short summary of the output in bullet point format below:

In moDCs..
- H1N1 inflicts strong transcriptonal changes..
- specifically upregulates genes involved in the immune response.
- Interferon stimulated genes (ISGs) are particularly upregulated.
- Polyl:C inflicts strong transcriptional changes..
- specifically upragulates genes involved in the immune response
- Overall high concordance between H1N1 and polyl:C treatment groups
 
These results were in large expected. It's of no surprise that virus infection leads to a transcriptional response in order to fight the infection.
In the same manner, polyl:C is a double stranded RNA mimic, and those, the cell will treat this as a potential viral infection and give a transcriptional response accordingly. 
 
However, for this project, the main question were on the computational side, where we wanted to compare the output from alignment-free methods and more traditional alignment-based methods. 
The idea came from the view that various people in the field had expressed, namely; that alignment is unnecessary for RNA-seq counting. Even further, that counting via aggregation of transcript
estimated counts to gene level counts would be more robust than simply mapping to gene level counts directly. At least to me, the mindset of looking at transcripts - which due to 
differential splicing and a vast amount of different isoforms can differ substantially in length and abundances - and then using these as input to gene level counting sounds attractive. 
There have been discussions in various community forums regarding the appropriateness of counting genes in this manner via alginment-free methods, and then using these counts as input to 
differential expression worflows based on the negative-binomial distribution. Some argue that these assumption does not hold, and therefore this should not be done, while other is of the opposite
side of the argument. In this project, we wanted to count the genes via three different methods (2 alignment-free: Kallisto, Salmon), and one alignment-based (STAR/HTSeq-count), 
and then use the input of gene-level counts derived from these methods into the same DESeq2 pipeline. 

Once again I will give a short summary of the output in bullet point format:

STAR/HTseq, Kallisto and Salmon ...
- All show high concordance in final output 
- Output distributions (p-values, mean dispersion variances) based on DESeq2 assumptions looks healthy 
- The two alignment-free methods display higher similarity in output 
- Some outlier differences were observed with unknown “truth” 
- Alignment-free methods works fast and run smoothly on a local machine 
- All methods ends up in the same biological conclusions for these data sets 

At least from our data and tests, I couldn't find any evidence that the Kallisto/Salmon gene-level counts as input to DESeq2 were inappropriate. The distributions from the output looks healthy,
and the conclusions are more or less the same as compared to the STAR/HTSeq-count pipeline. I guess that is the main take-away here, that all three methods ends up in the exact same biological conclusion.
There are some differences between the methods regarding the degree of up/down-regulation between samples. But the overall trend display very high concordance, it not like one up-regulated gene 
is found down-regulated in another pipeline. Instead, it's the degree of regulation that in some cases show some differences. 
While these differences in large are small, there were some examples which showed quite a big difference in output. I have shown such a example in an earlier entry.
The conclusion was that the differences in the abundances of isoforms could lead to large final output differences, an artefact of the fact that the mapping based method on gene level used here
do not resolve isoform differences. The question then becomes, is the calculated isoform abundance difference true in reality? Is the transcript identification so robust that you can
rely on its interpretation of your input data? That is not a trivial question to answer, and without a definite proof of the inherent true abundances, I cant confidently give the answer. 
Can you ever get a true benchmark? Apart from computational simulations, I guess one could make us of known spike-ins on real samples and see how the algorithms behave. 
On the other hand, this behaviour might differ a depending on the characteristics of the sequence, the degree of similarity between other sequences and motifs present in the transcriptome etc etc.
As for all things that we cannot see, and just make estimations on, there will always be uncertainty in the mix. In a simulation setting, you could of course benchmark against the truth, but then,
simulations are simulations, and the complexity and variations imposed on real data sets are hard to stimulate. 

One potential problem apart from any eventual uncertainty in the algorithms themselves is the use of reference transcriptomes. When using mapping-based methods, I have always gone with mapping against
the reference genome, and not transcriptome. This has in large been due to my take that the genome references are more consistent and well-annotated. When using a reference transcriptome,
one needs to ask - which transcriptome? What should I include and exclude? That questions directly infers a choice, and the more choices we have, the higher the risk for bias in our analysis.
Of course this really depends on the biological question at hand. Sometimes if might even be preferable to be able to fine-tune your reference.  
 
However, apart from this, the alignment-free methods seems to perform very well. The output looks like it should, and the idea of aggregating transcript-IDs to gene-level seems to work really well, 
at least for this data sets. I will continue with using these pipelines in my own research, maybe in parallel with the mapping approaches, to see how they behave on various different types of real data sets.

**poster**

The final output of the project is a poster. The project group will meet at a later date to look through this, but I have prepared a sketch on my idea: 

![poster]({{ site.url }}/assets/images/poster_wip_jb.PNG)

